front/ui/ui-charts/src/speedSpaceChart/components/common/SettingsPanel.tsx

```diff
@@ -76,6 +67,26 @@ const SettingsPanel = ({
                     [selection]: !prev.layersDisplay[selection],
                   },
                 }));
+
+                setChartHeight((prevHeight) => {
+                  let newHeight = prevHeight;
+                  if (selection === 'electricalProfiles') {
+                    newHeight +=
+                      LINEAR_LAYERS_HEIGHTS.ELECTRICAL_PROFILES_HEIGHT *
+                      (store.layersDisplay[selection] ? -1 : 1);
+                  }
+                  if (selection === 'powerRestrictions') {
+                    newHeight +=
+                      LINEAR_LAYERS_HEIGHTS.POWER_RESTRICTIONS_HEIGHT *
+                      (store.layersDisplay[selection] ? -1 : 1);
+                  }
+                  if (selection === 'speedLimitTags') {
+                    newHeight +=
+                      LINEAR_LAYERS_HEIGHTS.SPEED_LIMIT_TAGS_HEIGHT *
+                      (store.layersDisplay[selection] ? -1 : 1);
+                  }
+                  return newHeight;
+                });
```

Comment:
Please check the suggestion [below](https://github.com/OpenRailAssociation/osrd/pull/12594/files#r2298406345) about creating an `adjustHeightForLayerChange` function in the `SpeedSpaceChart` component first:
```suggestion
                adjustHeightForLayerChange(selection, store.layersDisplay[selection]);
```

https://github.com/OpenRailAssociation/osrd/pull/12594/files#r2298406345